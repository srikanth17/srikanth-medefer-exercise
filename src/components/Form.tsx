import React from 'react';
import { Field, InjectedFormProps, reduxForm } from 'redux-form';
import FormField from './FormField';
import formFields from '../utils/formFields';

interface FormProps extends InjectedFormProps {
  onFormSubmit: (values: any) => void;
}

const Form = (props: FormProps) => {
  const { handleSubmit, error, onFormSubmit } = props;

  return (
    <div>
      <form onSubmit={handleSubmit(onFormSubmit)}>
        {formFields.map(field => (
          <Field
            key={field.name}
            label={field.label}
            type="text"
            name={field.name}
            component={FormField}
          />
        ))}
        {error && <strong>{error}</strong>}
        <button type="submit" className="green btn-flat right white-text">
          Submit
          <i className="material-icons right">done</i>
        </button>
      </form>
    </div>
  );
};

export default reduxForm<{}, FormProps>({
  form: 'Form',
  destroyOnUnmount: false,
})(Form);
