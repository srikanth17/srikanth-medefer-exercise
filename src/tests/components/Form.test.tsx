import React from 'react';
import { shallow } from 'enzyme';
import Form from '../../components/Form';

test('should render Form correctly', () => {
  // @ts-ignore
  const wrapper = shallow(<Form />);
  expect(wrapper).toMatchSnapshot();
});

test('should not call onSubmit for empty form submission', () => {
  const onFormSubmit = jest.fn();
  // @ts-ignore
  const wrapper = shallow(<Form onFormSubmit={onFormSubmit} />);
  wrapper.find('form').simulate('submit');
  expect(onFormSubmit).not.toHaveBeenCalled();
});

test('should call onSubmit and submit form', () => {
  const firstName = 'your name';
  const surname = 'Surname';
  const email = 'test@test.com';
  const onFormSubmit = jest.fn();
  // @ts-ignore
  const wrapper = shallow(<Form onFormSubmit={onFormSubmit} />);
  wrapper.find('Field').at(0).simulate('change', firstName);
  wrapper.find('Field').at(1).simulate('change', surname);
  wrapper.find('Field').at(2).simulate('change', email);
  wrapper.find('form').simulate('submit');
  expect(onFormSubmit).toHaveBeenCalled();
});
