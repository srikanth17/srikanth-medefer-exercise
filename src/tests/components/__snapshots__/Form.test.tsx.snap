// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Form correctly 1`] = `
<Hoc
  destroyOnUnmount={false}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="Form"
  forwardedRef={null}
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  submitAsSideEffect={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
